### PR TITLE
Update Flatpak actions and runtime

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Closes #9451 since it superseeds it

1.Update Gtihub Action to fix the build-bundle option and bump its dependencies.
2.Update KDE Runtime to 6.5.

### Motivation and Context

1. Fix the build-bundle option which allow to avoid bundling for nothing and save time on workflow runs.
2. It can be done since docks are fixed with Qt 6.5
Also 2. Reduce the risk of people complaining about the EOL message when they install KDE 6.4 by installing OBS Studio through the CLI

### How Has This Been Tested?

1. I did many testing for the action while reworking aarch64 builds.
2. Not tested but since it should be fine since it's a minor bump and Windows and macOS are on 6.5

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

- CI

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: .github/workflows/build-project.yaml [hunk 1]
```diff
@@ -287,7 +287,7 @@ jobs:
           echo "cacheKey=${cache_key}" >> $GITHUB_OUTPUT
 
       - name: Build Flatpak Manifest 🧾
-        uses: flatpak/flatpak-github-actions/flatpak-builder@v6.1
+        uses: flatpak/flatpak-github-actions/flatpak-builder@1283416d7c0462e052ab0685d9444a7a48bfff3b
```
[the remark this answers is not in the document]
Comment (RytoEX) [reply .github/workflows/build-project.yaml:290]: 1283416d7c0462e052ab0685d9444a7a48bfff3b doesn't include 0955d7422f8703c835519c3ef38597c7965d7222 (Add kde-6.5) or its fixup add7abed5e5285f33477191af2d9a3e5f52ada01. Is that okay?

[the remark this answers is not in the document]
Comment (tytan652) [reply .github/workflows/build-project.yaml:290]: Those KDE 6.5 commits are meant for their image creation workflow which push the image to Bilal's docker hub registry.
It does nothing to the action itself, to get the new image we need to swap images to 6.5 which we do in the second commit.

[the remark this answers is not in the document]
Comment (RytoEX) [reply .github/workflows/build-project.yaml:290]: Got it.

**Review by RytoEX — APPROVED** [review]
Looks good to me.

**Pull request merged**